src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,491 @@
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepGetNumRhsEvals",
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  *sprk_storage = step_mem->method;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepPrintAllStats:
+
+  Prints integrator statistics
+  ---------------------------------------------------------------*/
+int SPRKStepPrintAllStats(void* arkode_mem, FILE* outfile, SUNOutputFormat fmt)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepPrintAllStats", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* step and rootfinding stats */
+  retval = arkPrintAllStats(arkode_mem, outfile, fmt);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (fmt)
+  {
+  case SUN_OUTPUTFORMAT_TABLE:
+    /* function evaluations */
+    fprintf(outfile, "f1 RHS fn evals              = %ld\n", step_mem->nf1);
+    fprintf(outfile, "f2 RHS fn evals              = %ld\n", step_mem->nf2);
+    break;
+  case SUN_OUTPUTFORMAT_CSV:
+    /* function evaluations */
+    fprintf(outfile, ",f1 RHS evals,%ld", step_mem->nf1);
+    fprintf(outfile, ",f2 RHS fn evals,%ld", step_mem->nf2);
+    break;
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, "ARKODE", "SPRKStepPrintAllStats",
+                    "Invalid formatting option.");
+    return (ARK_ILL_INPUT);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*===============================================================
+  SPRKStep parameter output
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  SPRKStepWriteParameters:
+
+  Outputs all solver parameters to the provided file pointer.
+  ---------------------------------------------------------------*/
+int SPRKStepWriteParameters(void* arkode_mem, FILE* fp)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int flag                   = 0;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepWriteParameters",
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* output ARKODE infrastructure parameters first */
+  flag = arkWriteParameters(ark_mem, fp);
+  if (flag != ARK_SUCCESS)
+  {
+    arkProcessError(ark_mem, ARK_MEM_NULL, "ARKODE::SPRKStep",
+                    "SPRKStepWriteParameters",
+                    "Error writing ARKODE infrastructure parameters");
+    return (flag);
+  }
+
+  /* print integrator parameters to file */
```

Comment:
Did you consider also writing out the SPRK method coefficients?

Reply:
We can, but we do not do this with the other steppers in the WriteParameters functions. Thoughts?

Reply:
It's really up to you for SPRKStep -- I think of these WriteParameters routines as purely being useful when debugging nasty issues.  If you add it here, then we can also consider adding those to ARKStep, ERKStep and MRIStep.

Reply:
Ill leave it out here as we can write them via the Butcher table functions.